src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -1808,56 +1736,45 @@ def list_ag_waf_rule_sets(client, _type=None, version=None, group=None):
 
 
 # region ApplicationGatewayWAFPolicy
-def create_ag_waf_policy(cmd, resource_group_name, policy_name,
-                         location=None, tags=None, rule_set_type='OWASP', rule_set_version='3.0'):
-    # https://docs.microsoft.com/en-us/azure/application-gateway/waf-overview
-
-    # mandatory default rule with empty rule sets
-    managed_rule_set = {
-        "rule_set_type": rule_set_type,
-        "rule_set_version": rule_set_version
-    }
-    managed_rule_definition = {
-        "managed_rule_sets": [managed_rule_set]
-    }
-
-    from .aaz.latest.network.application_gateway.waf_policy import Create
+class WAFCreate(_WAFCreate):
+    @classmethod
+    def _build_arguments_schema(cls, *args, **kwargs):
+        from azure.cli.core.aaz import AAZStrArg
+        args_schema = super()._build_arguments_schema(*args, **kwargs)
+        args_schema.rule_set_type = AAZStrArg(
+            options=["--type"],
+            help="Type of the web application firewall rule set.",
+            default="OWASP",
+            enum={"Microsoft_BotManagerRuleSet": "Microsoft_BotManagerRuleSet", "OWASP": "OWASP"},
+        )
+        args_schema.rule_set_version = AAZStrArg(
+            options=["--version"],
+            help="Version of the web application firewall rule set type, 0.1 is used for Microsoft_BotManagerRuleSet",
+            default="3.0",
+            enum={"0.1": "0.1", "2.2.9": "2.2.9", "3.0": "3.0", "3.1": "3.1", "3.2": "3.2"},
+        )
+        return args_schema
 
-    return Create(cli_ctx=cmd.cli_ctx)(command_args={
-        "resource_group": resource_group_name,
-        "name": policy_name,
-        "location": location,
-        "tags": tags,
-        "managed_rules": managed_rule_definition
-    })
+    def pre_operations(self):
+        args = self.ctx.args
+        managed_rule_set = {
+            "rule_set_type": args.rule_set_type,
+            "rule_set_version": args.rule_set_version
+        }
+        managed_rule_definition = {
+            "managed_rule_sets": [managed_rule_set]
+        }
```

Comment:
if not has_value(args.rule_set_type) or not has_value(args.rule_set_version), what's the managed_rule_definition?

Reply:
There is no case of that as both of them have default value.